### PR TITLE
Add limitcpu

### PR DESCRIPTION
**Pull request by mbargull**
This is "quite similar" to the `cpulimit` package. Submitting it here as a separate package since both projects diverged from another.
See http://limitcpu.sourceforge.net/ and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=843890#15.

cc @christopher-schroeder

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/limitcpu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  number: 0
+  skip: True  # [win]
+  script:
+    - make osx  # [osx]
+    - make install CFLAGS+=' -DVERSION=$(VERSION)'
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (scopatz) [inline recipes/limitcpu/meta.yaml:20]: make is also a build dependency

File: recipes/limitcpu/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+package:
+  name: limitcpu
+  version: {{ version }}
+
+source:
+  url: https://sourceforge.net/projects/limitcpu/files/limitcpu/cpulimit-{{ version }}.tar.gz
+  sha256: 2f8ceeb9de43a6de52ce50e67d618c715e714b72d806f3730b5661d5bfc9c075
+  patches:
+    - 0001-Fix-__APPLE_-typo.patch
```
Comment (mbargull) [inline recipes/limitcpu/meta.yaml:11]: Sent the patch upstream to the e-mail address mentioned at http://limitcpu.sourceforge.net/contact.php.
The OSX build still failed to the function `getjiffies` being only defined for Linux and FreeBSD.

**Comment by mbargull** [comment]
Thanks for the review @scopatz!

File: recipes/limitcpu/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  number: 0
+  skip: True  # [win or osx]
+  script:
+    - make install CFLAGS+=' -DVERSION=$(VERSION)'
```
Comment (scopatz) [inline recipes/limitcpu/meta.yaml:17]: Do you not need to run configure with the new PREFIX?

Comment (mbargull) [reply recipes/limitcpu/meta.yaml:17]: Nope, only a single `Makefile` that is really quiet pleasant for us and uses the environment's `CC`/`*FLAGS`/`PREFIX` as we need it.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**